apps/backend/src/Gump.WebApi/Controllers/RecipeController.cs

```diff
@@ -0,0 +1,246 @@
+		do
+		{
+			recipe = GetRecipe(recipeRepository.GetRandomId());
+		} while (
+			recipe is UnauthorizedResult ||
+			recipe is NotFoundObjectResult ||
+			recipe.Value.isPrivate
+		);
+
+		return Ok(((OkObjectResult)recipe).Value);
+	});
+
+	[AllowAnonymous]
+	[HttpGet("search")]
+	public IActionResult SearchRecipes([FromBody] SearchRecipeDto search) => this.Run(() =>
+	{
+		var searchResult = recipeRepository.Search(
+			search.SearchTerm,
+			search.Limit,
+			search.Offset,
+			search.AuthorId,
+			search.CategoryId
+		);
+
+		return Ok(searchResult.Select(r => new
+		{
+			id = r.Id,
+			title = r.Title,
+			author = r.AuthorId,
+			viewCount = r.ViewCount,
+			saveCount = r.SaveCount,
+			likeCount = r.Likes.Count,
+			isPrivate = r.IsPrivate
+		}));
+	});
+
+	[AllowAnonymous]
+	[HttpGet("today/{categoryId}")]
+	public IActionResult GetTodaysRecipe(ulong categoryId) => this.Run(() =>
+	{
+		RecipeModel randomRecipe;
+		do
+		{
+			ulong randomId = recipeRepository.GetRandomId(categoryId);
+			randomRecipe = recipeRepository.GetById(randomId);
+		} while (
+			randomRecipe.IsPrivate ||
+			randomRecipe.IsArchived ||
+			!randomRecipe.Tags.Any()
+		);
+
+		Random r = new();
+		string randomTag = randomRecipe.Tags[r.Next(randomRecipe.Tags.Count)];
+
+		return Ok(((OkObjectResult)SearchRecipes(new()
+		{
+			SearchTerm = randomTag,
+			Limit = 1,
+			Offset = 0,
+			CategoryId = categoryId
+		})).Value);
+	});
+
+	[HttpPost("create")]
+	public IActionResult CreateRecipe([FromBody] CreateRecipeDto recipe) => this.Run(() =>
+	{
+		var user = userRepository.GetById(ulong.Parse(User.Identity.Name));
+
+		RecipeModel newRecipe = new()
+		{
+			Title = recipe.Title,
+			AuthorId = user.Id,
+			Language = recipe.Language,
+			Serves = recipe.Serves,
+			Categories = recipe.Categories,
+			Tags = recipe.Tags,
+			Ingredients = recipe.Ingredients,
+			Steps = recipe.Steps,
+			IsOriginal = recipe.IsOriginal,
+			OriginalRecipeId = recipe.OriginalRecipeId,
+			IsPrivate = recipe.IsPrivate,
+			VisibleTo = recipe.VisibleTo
+		};
+
+		recipeRepository.Create(newRecipe);
+
+		return Ok(newRecipe.Id);
+	});
+
+	[HttpPatch("update")]
+	public IActionResult UpdateRecipe([FromBody] UpdateRecipeDto update) => this.Run(() =>
+	{
+		UserModel user = userRepository.GetById(ulong.Parse(User.Identity.Name));
+
+		if (update.Id != user.Id && !user.IsModerator)
+		{
+			return Forbid();
+		}
+
+		RecipeModel modifiedRecipe = recipeRepository.GetById(update.Id);
+
+		// if user has the same recipe, but with smaller id, then forbid
+		if (recipeRepository.GetAll().Any(x => x.Title == modifiedRecipe.Title && x.Id < modifiedRecipe.Id))
+		{
+			return Forbid();
+		}
+		
+		// if recipe is archived, then forbid
+		if (modifiedRecipe.IsArchived)
+		{
+			return Forbid();
+		}
+
+		// if user is moderator and not the author, then modify the original object
+		if (user.IsModerator && update.Id != user.Id)
+		{
+			modifiedRecipe = recipeRepository.GetById(update.Id);
```

Comment:
Ez meg mégis mi a jó isten? xd
Először is a newRecipe-t ez az if fölé kéne hozni, majd szimplán meg kéne hívni a repository update metódusát, ha moderátor és nem a sajátja. Returnölni egy okét és kész. Itt így nem történik abszolút semmi